test/test_cli.rb

```diff
@@ -1,441 +1,357 @@
 # frozen_string_literal: true
```

Comment:
Generally I considered `test_cli.rb` mostly like integration or system test, not unit one. That's why I prefer to not use stubs/mocks here to be confident in the future `cli.rb` refactoring.